codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsRuntimeConfig.java

```diff
@@ -94,12 +94,15 @@ public void addConfigInterfaceFields(
             writer.writeDocs("Enables FIPS compatible endpoints.")
                     .write("useFipsEndpoint?: boolean | __Provider<boolean>;\n");
         }
-        if (!settings.getExperimentalIdentityAndAuth()) {
-            if (isSigV4Service(settings, model)) {
+        if (isSigV4Service(settings, model) || isAwsService(settings, model)) {
+            if (!settings.getExperimentalIdentityAndAuth()) {
+                // This additional ID&A check is applied because
+                // the AwsSdkCustomizeSigV4Auth class also adds region
+                // under these conditions.
```

Comment:
So we want to write `region` independent of experimentalIdentityAndAuth (when isSigV4Service or isAwsService). With the documentation changing if it isAwsService (client will send requests) v/s just isSigV4Service (signing region). So it seems we can just do this from one place without looking at experimentalIdentityAndAuth.

Reply:
moving this to one location then